### PR TITLE
add base64 decode processor

### PR DESCRIPTION
**Pull request by ronshay**
This processor decodes base64 encoded values

### CODE REVIEW

File: sawmill-core/src/main/java/io/logz/sawmill/processors/Base64DecodeProcessor.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        fields.stream()
+                .filter(field -> !missingFields.contains(field))
+                .forEach(field -> {
+                    String value = doc.getField(field);
+                    doc.addField(field, new String(Base64.getDecoder().decode(value)));
+                });
+        return ProcessResult.success();
+    }
+
+    private boolean validateMissingFields(Set<String> missingFields) {
```
Comment (alexpalchuk) [inline sawmill-core/src/main/java/io/logz/sawmill/processors/Base64DecodeProcessor.java:43]: Would be a bit better to call it more `boolean` related. `areFieldsValid()` or something. WDYT?

**Review by alexpalchuk — APPROVED** [review]
Looks good 👍 . Just a small cosmetic comment

**Review by DanMelman — COMMENTED** [review]
1. Just curious, why do we provide the processor on a list of fields?
The common configuration of most processors usually gets one "source field" and one "target field". If you want to apply a processor on multiple fields, you need to define multiple processors/steps in your pipeline.
2. I see you decided to change the value in-place (override the original field value) rather than  setting a "target field". I wonder if this will make things harder to debug later, if some processing outputs bad decoding. Another option is to leave the original field as is, and if the user wants to override it then he just needs to add `removeField` and `renameField` processors to do that. If something goes wrong, he can omit the two additional steps for debugging.

**Comment by ronshay** [comment]
> 1. Just curious, why do we provide the processor on a list of fields?
>    The common configuration of most processors usually gets one "source field" and one "target field". If you want to apply a processor on multiple fields, you need to define multiple processors/steps in your pipeline.
> 2. I see you decided to change the value in-place (override the original field value) rather than  setting a "target field". I wonder if this will make things harder to debug later, if some processing outputs bad decoding. Another option is to leave the original field as is, and if the user wants to override it then he just needs to add `removeField` and `renameField` processors to do that. If something goes wrong, he can omit the two additional steps for debugging.

nice points. regarding the first one, i figured a single processor on multiple fields will probably run faster then n processors, but i do think it's a valid option.

about your second point, i totally agree. that will allow us to easily debug and the user can remove/rename as needed.
so i think we can go with keeping the original field, and setting a new "originalFieldName_decoded" field with the decoded value.
WDYT?

File: .github/workflows/git-secrets.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
   # This workflow contains a single job called "main"
   git-secrets:
     # The type of runner that the job will run on
-    runs-on: ubuntu-18.04
+    runs-on: macos-latest
```
Comment (alexpalchuk) [inline .github/workflows/git-secrets.yml:12]: can you clarify why you changed it? The OS can impact the way we sigh the code by codesigning command during new version release, it is sensitive

Comment (alexpalchuk) [inline .github/workflows/git-secrets.yml:12]: could you please clarify why we need to change it?

Comment (alexpalchuk) [reply .github/workflows/git-secrets.yml:12]: github stopped to support it

**Pull request merged**